snuba/web/views.py

```diff
@@ -98,14 +98,14 @@ def truncate_dataset(dataset: Dataset) -> None:
         clickhouse = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
         database = cluster.get_database()
 
-        tables_to_empty = {
-            schema.get_local_table_name()
-            for schema in storage.get_schemas().get_unique_schemas()
-            if isinstance(schema, TableSchema)
-        }
+        schema = storage.get_schema()
+
+        if not isinstance(schema, TableSchema):
+            return
+
+        table = schema.get_local_table_name()
 
-        for table in tables_to_empty:
-            clickhouse.execute(f"TRUNCATE TABLE IF EXISTS {database}.{table}")
+        clickhouse.execute(f"TRUNCATE TABLE IF EXISTS {database}.{table}")
```

Comment:
Is this not skipping the intnermediate schemas we were getting through 
```
for schema in storage.get_schemas().get_unique_schemas()
``` 
?

Reply:
Yeah, I actually changed this recently to also drop data from all schemas (including intermediate ones). But I've realised it's not necessary since they are just the materialized views. Dropping data from the underlying table they are referencing should be sufficient.